bin/build-locales

```diff
@@ -64,66 +64,82 @@ try {
     fs.mkdirSync(outputDir);
 }
 
-// message key with english string values (i.e. default values)
-var viewLocales = {};
-// FormattedMessage id with english string as value. Use for default values in translations
-// Sample structure: { 'general-general.blah': 'blah', 'about-about.blah': 'blahblah' }
-var idsWithICU = {};
-// reverse (i.e. english string with message key as the value) object for searching po files.
-// Sample structure: { 'blah': 'general-general.blah', 'blahblah': 'about-about.blah' }
-var icuWithIds = {};
-
 // get global locale strings first.
 var globalTemplateFile = path.resolve(__dirname, '../src/l10n.json');
-localeCompare.getIdsForView('general', globalTemplateFile, viewLocales, idsWithICU, icuWithIds);
+var ids = JSON.parse(fs.readFileSync(globalTemplateFile, 'utf8'));
 
+// message key with english string values (i.e. default values)
+var viewLocales = {
+    general: {
+        en: ids
+    }
+};
+var idsWithICU = localeCompare.idToICUMap('general', ids);
+var icuWithIds = localeCompare.icuToIdMap('general', ids);
+// localeCompare.getIdsForView('general', globalTemplateFile, viewLocales, idsWithICU, icuWithIds);
 
-// start with all views, and remove localized ones as they are iterated over
-var views = glob.sync(path.resolve(__dirname, '../src/views/*'));
-for (var i = 0; i < views.length; i++) {
-    views[i] = views[i].split('/').pop();
-}
+var views = [];
+var localizedAssetUrls = {};
 
-// get view-specific locale strings.
-var files = glob.sync(path.resolve(__dirname, '../src/views/**/l10n.json'));
-files.forEach(function (file) {
-    var dirPath = file.split('/');
-    dirPath.pop();
-    var view = dirPath.pop();
-    localeCompare.getIdsForView(view, file, viewLocales, idsWithICU, icuWithIds);
-});
+// start with all views, and remove localized ones as they are iterated over
+for (var v in routes) {
+    if (typeof routes[v].redirect !== 'undefined') {
+        continue;
+    }
+    
+    views.push(routes[v].name);
+    try {
+        var subdir = routes[v].view.split('/');
+        subdir.pop();
+        var l10n = path.resolve(__dirname, '../src/views/' + subdir.join('/') + '/l10n.json');
+        fs.accessSync(l10n, fs.F_OK);
+        ids = JSON.parse(fs.readFileSync(l10n, 'utf8'));
+        viewLocales[routes[v].name] = {
+            en: ids
+        };
+        idsWithICU = merge(idsWithICU, localeCompare.idToICUMap(routes[v].name, ids));
+        icuWithIds = merge(icuWithIds, localeCompare.icuToIdMap(routes[v].name, ids));
+    } catch (err) {
+        // check that routes' view path is not malformed, error if it is
+        try {
+            fs.accessSync(path.resolve(__dirname, '../src/views/' + routes[v].view + '.jsx'));
+        } catch (err) {
+            // the config for the view is not set up correctly, so throw the error.
+            throw err;
```

Comment:
It does determine if the l10n.json or view path exist, but the issue is that it's swallowing any error that isn't the one you're looking for. So if anything throws an error you don't expect — the errors you really want to know about — it silences them. That's a recipe for bugs that are hard to debug because the script will carry on like nothing happened but the results won't be what you expect.

So I think you should literally check if the file exists, and then decide to continue or not, not in a try...catch, so that any errors you don't expect get thrown and we'll know when something goes wrong.